server/data/data-cassandra/src/main/java/org/apache/james/user/cassandra/CassandraUsersDAO.java

```diff
@@ -212,11 +210,7 @@ public Mono<Void> addAuthorizedUsers(Username baseUser, Username userWithAccess)
             .setString(NAME, userWithAccess.asString())
             .setSet(DELEGATED_USERS, ImmutableSet.of(baseUser.asString()), String.class));
 
-        return getUserByNameReactive(userWithAccess).hasElement()
-            .filter(FunctionalUtils.identityPredicate())
-            .map(existAuthorizedUser -> (Statement) batchBuilder.build())
-            .switchIfEmpty(Mono.just(addAuthorizedStatement))
-            .flatMap(executor::executeVoid);
+        return executor.executeVoid(batchBuilder.build());
```

Comment:
Why this change?

Reply:
I explained it in the commit message:

GIVEN there are no users in user table
Behavior before: only delegatees source is modified, leaving the delegator source is not consistent with delegatees source

We have LDAP user repository edge case for delegation where LDAP does not depend on `user` table (therefore empty table), so we need to be more lenient managing delegations for this case.